shopelectro/request_data.py

```diff
@@ -0,0 +1,71 @@
+import typing
+
+from django import http
+from django_user_agents.utils import get_user_agent
+
+
+class RequestData:
+    def __init__(
+        self, request: http.HttpRequest, url_kwargs: typing.Dict[str, str]
+    ):
+        self.request = request
+        self.url_kwargs = url_kwargs
+
+
+class ProductList(RequestData):
+    """Data came from django urls to django views."""
+
+    PRODUCTS_ON_PAGE_PC = 48
+    PRODUCTS_ON_PAGE_MOB = 12
+    VIEW_TYPES = ['list', 'tile']
+
+    @property
+    def slug(self):
+        return self.url_kwargs.get('slug', '')
+
+    @property
+    def sorting_index(self):
+        return int(self.url_kwargs.get('sorting', 0))
+
+    @property
+    def tags(self) -> str:
+        """Tags list in url args format."""
```

Comment:
It ins't a clear comment. Let's remove it

Reply:
changed it to `""":return: 'color=red&size=large' for example."""`